lua/bufferline/options.lua

```diff
@@ -42,6 +53,17 @@ function options.closable()
   return get('closable', true)
 end
 
+--- @return bufferline.options.diagnostics
+function options.diagnostics()
+  local diagnostics = get('diagnostics', {})
+
+  if diagnostics.enable then
+    diagnostics = vim.tbl_extend('keep', diagnostics, {[ERROR] = 'Ⓧ ', [HINT] = '💡', [INFO] = 'ⓘ ', [WARN] = '⚠️ '})
```

Comment:
Using the actual `vim.diagnostics.severity` levels also simplifies the code quite a bit. It allows us to loop over the `vim.diagnostic.severity` to `count_diagnostics`, and then loop over `count_diagnostics` and check `options.diagnostics()` with the key from `count_diagnostics` to look at the icon being used for rendering.